opentracing-util/src/main/java/io/opentracing/util/ThreadLocalActiveSpanSource.java

```diff
@@ -36,7 +34,12 @@ public ThreadLocalActiveSpan activeSpan() {
 
     @Override
     public ActiveSpan makeActive(Span span) {
-        return new ThreadLocalActiveSpan(this, span, new AtomicInteger(1));
+        return new ThreadLocalActiveSpan(this, span, new AutoFinisher());
```

Comment:
@ivantopo @objectiser @yurishkuro 

To be clear, I am not proposing the current default because it makes migrations from 0.30 easier, I'm proposing it because I still think it's a sensible default. Though I'm getting the sense that others disagree?

The most common case for instrumentation (IMO) will still be a synchronous `try()` block, and the most concise syntax should auto-`finish()` the (Active)Span there IMO. There are other ways to accomplish this, of course... we could have other variants of `startXYZ()` along with `startActive()` and `startManual()`, for instance.

But I did want to make it clear that my thinking here is that the refcounting is still a net-good default, but just something that can be opted out of in cases where it's the wrong thing.

Thoughts?